### PR TITLE
Stats: Add version, frontend, and other non-dimensional info

### PR DESCRIPTION
**Pull request by Geod24**
This should help to get a feel of the system from Grafana.
Example output with `dub run -- -c doc/config.example.yaml`:

```
agora_application_info {version_="HEAD",build_timestamp="Fri Oct 30 19:30:51 2020",frontend_version="2094",public_key="GCOQEOHAUFYUAC6G22FJ3GZRNLGVCCLESEJ2AXBIJ5BJNUVTAERPLRIJ"} 1
```

The `HEAD` things needs to be fixed to refer to a commit, but for the time being, the build timestamp will suffice.

### CODE REVIEW

File: source/agora/node/FullNode.d [hunk 1]
```diff
@@ -209,9 +213,18 @@ public class FullNode : API
...
             this.pushBlock(this.params.Genesis);
 
+        Utils.getCollectorRegistry().addCollector(&this.collectAppStats);
         Utils.getCollectorRegistry().addCollector(&this.collectStats);
+        this.app_stats.setMetricTo!"agora_application_info"(
+            1,
+            "HEAD", // TODO: FIXME
+            __TIMESTAMP__, __VERSION__.to!string,
+            config.validator.enabled
+            ? config.validator.key_pair.address.toString() : null,
```
Comment (AndrejMitrovic) [inline source/agora/node/FullNode.d]:
```suggestion
                ? config.validator.key_pair.address.toString() : null,
```

it makes it clearer it's not a new argument.

File: source/agora/node/FullNode.d [hunk 2]
```diff
@@ -209,9 +213,18 @@ public class FullNode : API
         if (this.block_handlers.length > 0 && this.getBlockHeight() == 0)
             this.pushBlock(this.params.Genesis);
 
+        Utils.getCollectorRegistry().addCollector(&this.collectAppStats);
         Utils.getCollectorRegistry().addCollector(&this.collectStats);
+        this.app_stats.setMetricTo!"agora_application_info"(
+            1,
```
Comment (AndrejMitrovic) [inline source/agora/node/FullNode.d]: magic? unused?

**Review by AndrejMitrovic — APPROVED** [review]
nits

File: source/agora/node/FullNode.d (hunk 2 above)
Comment (Geod24) [reply source/agora/node/FullNode.d]: Yes

Comment (Geod24) [reply source/agora/node/FullNode.d]: (I'll add a comment)

**Comment by Geod24** [comment]
Comments addressed

**Review by AndrejMitrovic — APPROVED** [review]

**Pull request merged**